spec.bs

```diff
@@ -1881,6 +1928,178 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
+
+  1. Run the following steps [=in parallel=]:
+
+     1. Let |fencedFrameNonce| be |instance|'s [=fenced frame config instance/partition nonce=].
+
+     1. Let |credentiallessNonce| be
+
+     Issue: the page credentialless nonce
+     (<a href="https://github.com/WICG/fenced-frame/issues/191">WICG/fenced-frame#191</a>)
+
+     1. Invoke [=revoke network for a partition nonce=] on |fencedFrameNonce|.
+
+     1. Invoke [=revoke network for a partition nonce=] on |credentiallessNonce|.
+
+     1. Set |instance|'s [=fenced frame config instance/untrusted network status=] to [=untrusted
+        network status/disabled for this tree=].
+
+     1. [=list/Append=] |p| to |instance|'s [=fenced frame config instance/on network disabled
+        promises=].
+
+     1. [=Recalculate the untrusted network status of all fenced frame descendants=] given
+        |global|'s [=Window/browsing context=]'s [=browsing context/top-level traversable=].
+
+  1. Return |p|.
+</div>
+
+<div algorithm>
+  To <dfn>Recalculate the untrusted network status of all fenced frame descendants</dfn> given a
+  [=top-level traversable=] |topLevelTraversable|, run these steps:
+
+  1. [=Assert=]: this is running [=in parallel=].
+
+  1. Let |navigables| be |topLevelTraversable|'s [=navigable/active document=]'s
+     [=Document/inclusive descendant navigables=] with [=inclusive-dn-unfenced|unfenced=] set to
+     true.
+  
+  1. Let |navigablesWithNetworkChildren| be an empty [=set=].
+
+  1. [=iteration/While=] |navigables| is not [=queue/empty=]:
+
+     1. Let |currentNavigable| be the result of [=queue/dequeue|dequeuing=] from |navigables|.
+
+     1. If |currentNavigable| is not a [=fenced navigable container/fenced navigable=], then
+        [=iteration/continue=].
+
+     1. Let |config| be |currentNavigable|'s [=navigable/active browsing context=]'s [=browsing
+        context/fenced frame config instance=].
+
+     1. If |config| is null, then [=iteration/continue=].
+
+     1. If |config|'s [=fenced frame config instance/untrusted network status=] is [=untrusted
+        network status/disabled for this and descendant trees=], then [=iteration/continue=].
+
+     1. Let |networkCutoffReady| be true if |navigablesWithNetworkChildren| does not [=set/contain=]
+        |currentNavigable| and |config|'s [=fenced frame config instance/untrusted network status=]
+        is [=untrusted network status/disabled for this tree=], false otherwise.
+
+     1. If |networkCutoffReady| is true:
+
+        1. Set |config|'s [=fenced frame config instance/untrusted network status=] to [=untrusted
+           network status/disabled for this and descendant trees=].
+
+        1. Let |global| be |currentNavigable|'s [=navigable/active window=].
+
+        1. For each |promise| in |config|'s [=fenced frame config instance/on network disabled
+           promises=]:
+
+           1. [=Queue a global task=] on the [=DOM manipulation task source=] given |global|, to
```

Comment:
I'm a little worried about this. `|global|` comes from the current nav's active window, but is it guaranteed that `|promise|` was constructed _within_ that global? For example, what if inside a fenced frame, a cross-origin descendant iframe calls `disableUntrustedNetwork()`. First, is that allowed? Second, if it's allowed, then I think we'd end up scheduling the resolution of that Promise (that was created in the cross-origin descendant global) on the global of its nearest fenced frame root (since at this point `|currentNavigable|` is guaranteed to be a fenced navigable's global).

Does this make sense / am I right about this being possible?

Reply:
> For example, what if inside a fenced frame, a cross-origin descendant iframe calls disableUntrustedNetwork(). First, is that allowed?

No. `disableUntrustedNetwork()` throws a `TypeError` if the origin that calls the method is cross-origin to the mapped URL of the ff config instance. The promises can only come from either the fenced frame root or a same-origin descendant.

Reply:
Well, same-origin descendant iframes have different globals, but we resolve them here on the fenced frame's global, which is bad, right?